File: CS/PersistentRules.Module/Updater.cs
using System;
using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.BaseImpl;
using MainDemo.Module;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Validation;

namespace PersistentRules.Module {
    public class Updater : ModuleUpdater {
        public Updater(IObjectSpace objectSpace, Version currentDBVersion) : base(objectSpace, currentDBVersion) { }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();

            Person Sam = ObjectSpace.CreateObject<Person>();
            Sam.FirstName = "Sam";
            Sam.Save();

            RuleRequiredFieldPersistent LastNameIsNotEmpty = ObjectSpace.CreateObject<RuleRequiredFieldPersistent>();
            LastNameIsNotEmpty.ObjectTypeCore = typeof(Person);
            LastNameIsNotEmpty.Property = "LastName";
            LastNameIsNotEmpty.ContextIDs = "Save";
            LastNameIsNotEmpty.InvertResult = false;
            LastNameIsNotEmpty.Id = "Person_LastNameIsNotEmpty";
            LastNameIsNotEmpty.SkipNullOrEmptyValues = false;
            LastNameIsNotEmpty.RuleName = LastNameIsNotEmpty.Id;
            LastNameIsNotEmpty.Save();
        }
    }
}

File: VB/PersistentRules.Module/Updater.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.BaseImpl
Imports MainDemo.Module

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Validation

Namespace PersistentRules.Module
	Public Class Updater
		Inherits ModuleUpdater
		Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
			MyBase.New(objectSpace, currentDBVersion)
		End Sub
		Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
			MyBase.UpdateDatabaseAfterUpdateSchema()

			Dim Sam As Person = ObjectSpace.CreateObject(Of Person)()
			Sam.FirstName = "Sam"
			Sam.Save()

			Dim LastNameIsNotEmpty As RuleRequiredFieldPersistent = ObjectSpace.CreateObject(Of RuleRequiredFieldPersistent)()
			LastNameIsNotEmpty.ObjectTypeCore = GetType(Person)
			LastNameIsNotEmpty.Property = "LastName"
			LastNameIsNotEmpty.ContextIDs = "Save"
			LastNameIsNotEmpty.InvertResult = False
			LastNameIsNotEmpty.Id = "Person_LastNameIsNotEmpty"
			LastNameIsNotEmpty.SkipNullOrEmptyValues = False
			LastNameIsNotEmpty.RuleName = LastNameIsNotEmpty.Id
			LastNameIsNotEmpty.Save()
		End Sub
	End Class
End Namespace
